### PR TITLE
Adding usage documentation

### LINKED ISSUE #37: Lack of documentation for command-line arguments
This tool contains many useful and powerful features, but none of these are documented. Documentation on how to use this tool to its full potential needs to be created to make it easier for our clinical partners to work with their datasets.

### PR DESCRIPTION
**Pull request by ant0nsc**
Closes #37

### CODE REVIEW

File: commandline_args.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+The spacings to use for geometric normalization, in millimeters.
+3 values must be provided, semicolon separated, in the order X; Y; Z.
+Using a spacing of 0 in any dimension means to not change this specific dimension.
+
+Example: `--geoNorm 1;1;0` will normalize the spacing to 1mm in the X and Y dimensions, and not change the third dimension.
+
+## `--groundTruthDescendingPriority`: Controls the order of structures when making the dataset mutually exclusive
+
+The priority mapping for structures, to ensure they are mutually exclusive. Only structures with names in this set will be uded in the
+resulting dataset. Special case: if names are prefixed by "+", structures with those names will be included, but mutual usion will not be
```
Comment (peterhessey) [inline commandline_args.md:51]:
```suggestion
resulting dataset. Special case: if names are prefixed by "+", structures with those names will be included, but mutual inclusion will not be
```

Typo? Not sure if this was the right word

File: commandline_args.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
+# Commandline options for InnerEye-CreateDataset
+
+The list below contains the main commandline switches for InnerEye-CreateDataset. For a full list of options, compile
+the `InnerEye.CreateDataset.Runner` project and run `InnerEye.CreateDataset.Runner.exe dataset --help`.
+
+The options are defined in [CommandlineCreateDataset.cs](https://github.com/microsoft/InnerEye-CreateDataset/blob/main/Source/projects/InnerEye.CreateDataset.Core/Commandline/CommandlineCreateDataset.cs).
+
+## `--rename`: Renaming and modifying ground truth structures
+
+The argument value is a mapping of structure names in the Dicom dataset to structure names in the nifti dataset.
```
Comment (peterhessey) [inline commandline_args.md:10]:
```suggestion
The argument value is a mapping of structure names in the DICOM dataset to structure names in the NIFTI dataset.
```

Comment (peterhessey) [reply commandline_args.md:10]: Inconsistency in case on these acronyms throughout the file

File: commandline_args.md [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+## `--discardInvalidSubjects`
+
+If this switch is provided, invalid subjects will be discarded, and the dataset
+will be created without them. If not provided, the program will exit when encountering the first invalid subject.
+If you set this switch, you should check the output of the job for information about discarded subjects.
+
+## `--allowNameClashes`
+
+If this switch is provided (True), structure renaming causes a choice to be made between named structures when several options exist, er than causing
```
Comment (peterhessey) [inline commandline_args.md:72]: "er than causing" - typo?

File: commandline_args.md [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+renaming to fail. Specifically, for a name mapping "oldName1,oldName2,...:newName":
+
+- If AllowNameClashes is not set and structures exist for more than one of the names (old and/or new), renaming s.
```
Comment (peterhessey) [inline commandline_args.md:75]: "renaming s." - what is "s"?

File: commandline_args.md [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+- Otherwise, if a structure named newName exists, no renaming is done.
+- Otherwise, if a structure named oldNameK exists and no structures with names oldName1, ..., oldName[K-1] t, then it is
+renamed to newName.
+
+Thus in summary, if names clash, an existing "newName" structure is preferred (we assume the oldNames are -preferred variants
```
Comment (peterhessey) [inline commandline_args.md:80]:
```suggestion
Thus in summary, if names clash, an existing "newName" structure is preferred (we assume the oldNames are preferred variants
```

typo?

**Review by peterhessey — CHANGES_REQUESTED** [review]

File: commandline_args.md (hunk 4 above)
Comment (ant0nsc) [reply commandline_args.md:75]: oops. "fails" :-)

**Review by peterhessey — APPROVED** [review]

**Pull request merged**